gnd/src/main/java/com/google/android/gnd/ui/offlinearea/OfflineAreasViewModel.java

```diff
@@ -16,39 +16,57 @@
 
 package com.google.android.gnd.ui.offlinearea;
 
+import static java8.util.stream.StreamSupport.stream;
+
+import android.content.Context;
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.LiveDataReactiveStreams;
 import com.google.android.gnd.model.basemap.OfflineArea;
 import com.google.android.gnd.repository.OfflineAreaRepository;
+import com.google.android.gnd.system.GeocodingManager;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.ui.common.Navigator;
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import dagger.hilt.android.qualifiers.ApplicationContext;
+import java8.util.stream.Collectors;
 import javax.inject.Inject;
 
 /**
  * View model for the offline area manager fragment. Handles the current list of downloaded areas.
  */
 public class OfflineAreasViewModel extends AbstractViewModel {
 
-  private LiveData<ImmutableList<OfflineArea>> offlineAreas;
+  private LiveData<ImmutableMap<String, OfflineArea>> offlineAreas;
   private final Navigator navigator;
+  private final GeocodingManager geocoder;
 
   @Inject
-  OfflineAreasViewModel(Navigator navigator, OfflineAreaRepository offlineAreaRepository) {
+  OfflineAreasViewModel(
+      Navigator navigator,
+      OfflineAreaRepository offlineAreaRepository,
+      @ApplicationContext Context context) {
     this.navigator = navigator;
+    this.geocoder = new GeocodingManager(context);
```

Comment:
Instead of creating an instance here, how about converting `GeocodingManager` into an injectable dependency?

Reply:
+1. I believe all you need is to add `@Inject` to the constructor.

Reply:
56b49e5